### PR TITLE
docs/nia: Add configuration overview and move config page to top level

### PR DESCRIPTION
**Pull request by findkim**
* Moved configuration page to top level nav + redirect from old path: [/docs/nia/configuration](https://deploy-preview-9408--consul-docs-preview.netlify.app/docs/nia/configuration#backend)
* Added new page Get Started > [Configure](https://deploy-preview-9408--consul-docs-preview.netlify.app/docs/nia/installation/configure) to capture CTS config file at a high level and explain the main components
* Minor updates to include make it clear the requirement of `driver.terraform.required_provider`
* Updated [Terraform backend](https://deploy-preview-9408--consul-docs-preview.netlify.app/docs/nia/configuration#backend) to include newly supported state stores

### CODE REVIEW

File: website/_redirects [hunk 1]
```diff
@@ -394,6 +394,8 @@
 /docs/platform/k8s/uninstalling                     /docs/k8s/operations/upgrading                                                     301!
 /docs/platform/k8s/*                                /docs/k8s/:splat                                                                   301!
 
+/docs/nia/installation/configuration                /docs/nia/configuration                       301!
+
```
Comment (lornasong) [inline website/_redirects:398]: nice! tested this out and confirmed the redirect

File: website/pages/docs/nia/installation/configure.mdx [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+}
+```
+
+## Terraform Providers
+
+Configuring Terraform providers within Consul-Terraform-Sync requires 2 config components. The first is the [`terraform_provider` block](/docs/nia/configuration#terraform-provider). This block resembles [provider blocks for Terraform configuration](https://www.terraform.io/docs/configuration/providers.html) and has the same responsibility for understanding API interactions and exposing resources for a specific infrastructure platform.
+
+Terraform modules configured for task automation may require configuring the referenced providers. For example, to configure the host address and authentication to interface with your network infrastructure. Refer to the Terraform provider documentation hosted on the [Terraform Registry](https://registry.terraform.io/browse/providers) to find available options. The `terraform_provider` block is loaded by Consul-Terraform-Sync during runtime and processed to be included in [autogenerated Terraform configuration files](/docs/nia/network-drivers#provider) used for task automation. Omitting the `terraform_provider` block for a provider will defer to the Terraform behavior assuming an empty default configuration.
```
Comment (lornasong) [inline website/pages/docs/nia/installation/configure.mdx:36]: > For example, to configure the host address and authentication to interface with your network infrastructure.

This reads like a phrase to me. Is there more maybe? e.g. "to configure..., you will need to do x"

**Review by lornasong — APPROVED** [review]
🎉  just left one comment about a sentence

File: website/pages/docs/nia/installation/configure.mdx (hunk 2 above)
Comment (findkim) [reply website/pages/docs/nia/installation/configure.mdx]: Ah true true, I don't have the "X" in mind, so maybe I'll simplify it

> For example, configuring the host address and authentication to interface with your network infrastructure.

**Comment by findkim** [comment]
Thanks for the review @lornasong ! I reread it this morning and made a few more changes if you don't mind another read through 🙏

File: website/pages/docs/nia/installation/configure.mdx (hunk 2 above)
Comment (lornasong) [reply website/pages/docs/nia/installation/configure.mdx]: Yes! Gotcha. Using 'configuring' parallels with the previous sentence so I get what the example is related to.

**Review by lornasong — APPROVED** [review]
looks good!

**Pull request merged**